### DISCUSSION TITLE
Assembler Support in SLEIGH C++

### DISCUSSION THREAD

**Discussion by umarfarook882** [NationalSecurityAgency/ghidra#7022] [Q&A]
Currently Ghidra provide SLEIGH C++ Interface ([here](https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Features/Decompiler/src/decompile/cpp)) as library, which provide two major components disassembly and decompilation engines. This help researcher to build library for analysis without requiring Ghidra Installation 

Question:

1. Any plan in future to support Assembler component in SLEIGH C++ interface?
2. In Ghidra Java API,  [Assembler interface](https://ghidra.re/ghidra_docs/api/ghidra/app/plugin/assembler/Assembler.html) is provided as plugin ([here](https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/app/plugin/assembler)), is it possible run Assembler without ghidra installation via [jpype](https://jpype.readthedocs.io/en/latest)?

Note: Looked at [PyGhidra](https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Features/PyGhidra/src/main/py) it required ghidra installation to access Ghidra API

**Comment by ryanmkurtz** [comment]
> In Ghidra Java API, [Assembler interface](https://ghidra.re/ghidra_docs/api/ghidra/app/plugin/assembler/Assembler.html) is provided as plugin ([here](https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Framework/SoftwareModeling/src/main/java/ghidra/app/plugin/assembler)), is it possible run Assembler without ghidra installation via [jpype](https://jpype.readthedocs.io/en/latest)?
Note: Looked at [PyGhidra](https://github.com/NationalSecurityAgency/ghidra/tree/master/Ghidra/Features/PyGhidra/src/main/py) it required ghidra installation to access Ghidra API

No, PyGhidra/JPype requires a full Ghidra installation to be present.

**Reply by umarfarook882** [reply]
Any plan in future to support Assembler component in SLEIGH C++ interface?

**Reply by ryanmkurtz** [reply]
No, that has never been discussed, AFAIK.
